src/uu/cksum/src/cksum.rs

```diff
@@ -188,21 +202,30 @@ fn digest_read<T: Read>(
     let mut digest_writer = DigestWriter::new(digest, true);
     let output_size = std::io::copy(reader, &mut digest_writer)? as usize;
     digest_writer.finalize();
-
     if digest.output_bits() > 0 {
-        Ok((digest.result_str(), output_size))
+        Ok((digest.result_bytes(), output_size))
     } else {
         // Assume it's SHAKE.  result_str() doesn't work with shake (as of 8/30/2016)
         let mut bytes = Vec::new();
         bytes.resize((output_bits + 7) / 8, 0);
         digest.hash_finalize(&mut bytes);
-        Ok((encode(bytes), output_size))
+        Ok((bytes, output_size))
+    }
+}
+
+fn reverse_cksum(checksum: &[u8]) -> Vec<u8> {
+    let sz = checksum.len();
+    let mut ret: Vec<u8> = std::iter::repeat(0).take(sz).collect::<Vec<u8>>();
+    for i in 0..sz {
+        ret[i] = checksum[sz - i - 1].reverse_bits();
     }
+    ret
 }
```

Comment:
Hmm, I'd write this without a for loop. 

```rust 
fn reverse_cksum(checksum: &[u8]) -> Vec<u8> {
    checksum.iter().rev().map(|&b| b.reverse_bits()).collect()
}

```

Reply:
@tertsdiepraam I forgot whether we try to avoid for loops for uutils, is this a rule here?

Reply:
Just use whatever you think is most clear in each situation!

Reply:
@howjmay what do you think here?